src/Tribe/Rewrite.php

```diff
@@ -426,9 +426,6 @@ public function get_canonical_url( $url, $force = false ) {
 		$query         = (string) parse_url( $url, PHP_URL_QUERY );
 		wp_parse_str( $query, $query_vars );
 
-		// Drop any query var that is not a scalar; it should not be handled.
-		$query_vars = array_filter( $query_vars, 'is_scalar' );
-
 		if ( isset( $query_vars['paged'] ) && 1 === (int) $query_vars['paged'] ) {
```

Comment:
What about doing recursion and going through a multidimensional array and check if the value is_scalar? @bordoni and I were discussing that idea for my solution too. Something similar to this method: 
https://github.com/moderntribe/tribe-common/blob/924035eadc6b90283e772ff62ebd411e74f40ea8/src/Tribe/Utils/Array.php#L210